Cover mTLS (#556)
kubewarden/docs: docs/reference/security-hardening/webhook-mtls.md

## Block External Traffic Using Network Policies

The webhooks are only expected to accept requests from the Kubernetes API server and from the audit scanner component. By default, however, the webhooks
can accept traffic from any source.
If you are using a CNI that supports Network Policies, you can create a policy that blocks traffic that doesn't originate from the API server.

The built-in NetworkPolicy resource in Kubernetes can't block or admit traffic from the cluster hosts, and the `kube-apiserver` process is always running on the host network.
Therefore, you must use the advanced network policy resources from the CNI in use. Examples for Calico and Cilium follow. Consult the documentation for your CNI for more details.

### Calico

Use the NetworkPolicy resource in the `crd.projectcalico.org/v1` API group, define a network policy like the following one:

```yaml
apiVersion: crd.projectcalico.org/v1
kind: NetworkPolicy
metadata:
  name: allow-k8s-and-audit-scanner
  namespace: kubewarden # namespace where the kubewarden stack is deployed
spec:

>   selector: 'app.kubernetes.io/name == "kubewarden-controller" || has(kubewarden/policy-server)'

Reviewer: Should we update to the new labels already, with a `managed-by`?

Reviewer 2: Yeah, I agree. As we are about to merge the [PR](https://github.com/kubewarden/kubewarden-controller/pull/1037) adding the recommended policies, I think we can hold this PR until we merge that one.

        selector:
          'app.kubernetes.io/name == "kubewarden-controller" || has(kubewarden/policy-server)'
```

### Cilium

Use the CiliumNetworkPolicy resource in the `cilium.io/v2` API group, define a network policy like the following one:

```yaml
apiVersion: "cilium.io/v2"
kind: CiliumNetworkPolicy
metadata:
  name: allow-k8s-and-audit-scanner
  namespace: kubewarden # namespace where the kubewarden stack is deployed
spec:
  endpointSelector:
    matchExpressions:
      - key: app.kubernetes.io/name
        operator: In
        values:
          - kubewarden-controller
      - key: kubewarden/policy-server
        operator: Exists
  ingress:
    # this allows the Kubernetes API server to reach the kubewarden controller and
    # all the policy server instances
    - fromEntities:
      - host
      - remote-node
    # this allows all the workloads defined inside of the kubewarden namespace
    # (including audit-scanner) to reach the kubewarden controller and all the
    # policy server instances
    - fromEndpoints:
        - matchLabels:
            # namespace where the kubewarden stack is deployed
            k8s:io.kubernetes.pod.namespace: kubewarden
```

> ## Require the Kubernetes API Server to Authenticate to the Webhook

Reviewer 2: As this section is pretty similar to the on in the 'howto' page, shouldn't we mention that page in case the user want to see a full example on how to enable it?

Author: Good point, I've added a `tip` section that contains a reference to the howto

---
sidebar_label: Webhook Mutual TLS
title: Secure webhooks with mutual TLS
description: Harden the webhook configuration.
keywords: [kubewarden, kubernetes, security]
doc-persona: [kubewarden-operator, kubewarden-integrator]
doc-type: [reference]
doc-topic: [reference, security]
---

<head>
  <link rel="canonical" href="https://docs.kubewarden.io/reference/security-hardening/webhook-mtls"/>
</head>

> The Kubewarden stack uses different webhooks to enforce policies on the Kubernetes cluster. Each `PolicyServer` instance exposes a webhook

Reviewer 3: ```suggestion
The Kubewarden uses utilizes various webhooks to enforce policies within the Kubernetes cluster. Each `PolicyServer` instance exposes a webhook
```
